tuxedo-core/src/constraint_checker.rs

```diff
@@ -99,14 +188,19 @@ pub mod testing {
     impl SimpleConstraintChecker for TestConstraintChecker {
         type Error = ();
 
+        type Accumulator = ();
+
         fn check(
             &self,
             _input_data: &[DynamicallyTypedData],
             _peek_data: &[DynamicallyTypedData],
             _output_data: &[DynamicallyTypedData],
-        ) -> Result<TransactionPriority, ()> {
+        ) -> Result<ConstraintCheckingSuccess<()>, ()> {
             if self.checks {
-                Ok(0)
+                Ok(ConstraintCheckingSuccess {
+                    priority: 0,
+                    accumulator_value: (),
+                })
```

Comment:
I guess repeating this is kind of annoying. Maybe we should have an impl like

```rust
impl From<TransactionPrtiority> for ConstraintCheckingSuccess<()> {

  fn from(priority: TransactionPriority) -> Self {
    Self {
      priority,
      accumulator_value: (),
    }
  }
}
```

Reply:
Haha, I am flattered by the credits. :)

I can definitely recommend doing these boilerplate implementations - 0 cost abstractions; hence, 0 reason not to do so, haha.